src/Microsoft.DotNet.Build.Tasks/PackageFiles/tests.targets

```diff
@@ -206,6 +209,28 @@
       OutputListLocation="$(OutDir)assemblylist.txt"
       Condition="'$(UseLegacyXunitPerfRunner)'=='true'"
      />
+     
+     <!-- For Desktop Tests, we need to generate Binding Redirects. -->
+     <ItemGroup>
+        <_FileNamesInReferenceCopyLocalPaths Include="@(ReferenceCopyLocalPaths->'%(Filename)%(Extension)')">
+          <SourcePath>%(ReferenceCopyLocalPaths.Identity)</SourcePath>
+        </_FileNamesInReferenceCopyLocalPaths>
+        <_FileNamesInTestCopyLocalByFileNameWithoutDuplicates Include="@(_TestCopyLocalByFileNameWithoutDuplicates->'%(Filename)%(Extension)')">
```

Comment:
Isn't this 
```
+        <_FileNamesInTestCopyLocalByFileNameWithoutDuplicates Include="@(_TestCopyLocalByFileNameWithoutDuplicates->'%(Filename)%(Extension)')">
 +          <SourcePath>%(_TestCopyLocalByFileNameWithoutDuplicates.SourcePath)</SourcePath>
 +        </_FileNamesInTestCopyLocalByFileNameWithoutDuplicates>
```

the same as 
```
<_FileNamesInTestCopyLocalByFileNameWithoutDuplicates  Include="@(_TestCopyLocalByFileNameWithoutDuplicates)"/>
```

ie., metadata like `SourcePath` should be copied automatically when you populate one list from anotehr using an item transform (an @ expression)